File: frame/src/main/java/com/ssz/frame/base/BaseActivity.java
package com.ssz.frame.base;

import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import android.support.annotation.Nullable;


/**
 * @author : zsp
 * time : 2019 08 2019/8/23 9:36
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        initView();
        initData();
        setEvent();
    }
    protected abstract int getLayoutId();
    protected abstract void initView();
    protected abstract void initData();
    protected abstract void setEvent();

    /**
     * 以下改动目的：字体大小不随系统字体大小的改变而改变
     * @param newConfig
     */

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        if (newConfig.fontScale != 1) { //fontScale不为1，需要强制设置为1
            getResources();
        }
    }

    @Override
    public Resources getResources() {
        Resources resources = super.getResources();
        if (resources.getConfiguration().fontScale != 1) { //fontScale不为1，需要强制设置为1
            Configuration newConfig = new Configuration();
            newConfig.setToDefaults();//设置成默认值，即fontScale为1
            resources.updateConfiguration(newConfig, resources.getDisplayMetrics());
        }
        return resources;
    }


}

File: frame/src/main/java/com/ssz/frame/base/BaseActivity.kt
package com.ssz.frame.base

import android.annotation.SuppressLint
import android.app.Dialog
import android.content.Intent
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.widget.PopupWindow
import android.widget.TextView
import com.ssz.frame.utils.toast.ToastUtil

/**
 * @author : zsp
 * time : 2019 09 2019/9/18 15:33
 */
@SuppressLint("Registered")
abstract class BaseActivity : AppCompatActivity(){

    protected fun getText(view: TextView) = view.text.toString()

    /**
     * 获取数字，空默认为0
     */
    protected fun getTextToInt(view: TextView):Int{
        val str = view.text.toString()
        return if (str.isEmpty()) 0 else str.toInt()
    }

    protected fun dismiss(dialog: Dialog) {
        if (dialog.isShowing) dialog.dismiss()
    }

    protected fun dismiss(popupWindow: PopupWindow) {
        if (popupWindow.isShowing) popupWindow.dismiss()
    }

    fun showToast(msg: String) {
        ToastUtil.showToast(this, msg)
    }

    fun showToast(msg: String, gravity: Int) {
        ToastUtil.showToast(this, msg, gravity)
    }

    /*************************** get resources *************************/

    fun getColorById(res : Int): Int{
       return ContextCompat.getColor(this,res)
    }

    fun getDrawableById(res : Int): Drawable? {
        return ContextCompat.getDrawable(this,res)
    }

    fun getDimenById(dimenId : Int):Int{
        return resources.getDimension(dimenId).toInt()
    }

    fun getDimenFloatById(dimenId : Int):Float{
        return resources.getDimension(dimenId)
    }

    fun getStringById(resId : Int):String{
        return resources.getString(resId)
    }

    /*************************** get resources end *************************/

    /**
     * 跳转页面
     *
     * @param clz 所跳转的目的Activity类
     */
    fun startActivity(clz: Class<*>) {
        startActivity(Intent(this, clz))
    }

    /**
     * 跳转页面
     *
     * @param clz         所跳转的Activity类
     * @param requestCode 请求码
     */
    fun startActivityForResult(clz: Class<*>, requestCode: Int) {
        startActivityForResult(Intent(this, clz), requestCode)
    }

    /**
     * 跳转页面
     *
     * @param clz    所跳转的目的Activity类
     * @param bundle 跳转所携带的信息
     */
    fun startActivity(clz: Class<*>, bundle: Bundle?) {
        val intent = Intent(this, clz)
        if (bundle != null) {
            intent.putExtra("bundle", bundle)
        }
        startActivity(intent)
    }

    /**
     * 跳转页面
     *
     * @param clz         所跳转的Activity类
     * @param bundle      跳转所携带的信息
     * @param requestCode 请求码
     */
    fun startActivityForResult(clz: Class<*>, requestCode: Int, bundle: Bundle?) {
        val intent = Intent(this, clz)
        if (bundle != null) {
            intent.putExtra("bundle", bundle)
        }
        startActivityForResult(intent, requestCode)
    }

}